lms/djangoapps/sga_api/views.py

```diff
@@ -0,0 +1,293 @@
+    ):
+        courses = {}
+        for course_id in list_of_courses_id:
+            results = self._enroll_unenrol_users_for_course(
+                course_id, list_of_users_email, request,
+                action, email_students, auto_enroll
+            )
+            courses[course_id] = self._make_result_dict_for_course(
+                 results, action, auto_enroll
+            )
+        return courses
+
+    def _enroll_unenrol_users_for_course(
+            self, course_id, list_of_users_email, request,
+            action='enroll', email_students=False, auto_enroll=False
+    ):
+        result_list = []
+        for users_email in list_of_users_email:
+    
+            try:
+                user = get_student_from_identifier(users_email)
+                email = user.email
+                language = get_user_email_language(user)
+            except User.DoesNotExist:
+                if auto_enroll and action == 'enroll':
+                    email = users_email
+                    language = None
+                else:
+                    result_list.append(
+                        self._make_result_dict_for_identifier(
+                            users_email=users_email,
+                            error='User with {} email does not exist'.format(users_email),
+                            after={}, before={}
+                        )
+                    )
+                    continue
+            try:
+                course_id_obj = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+                course = modulestore().get_course(course_id_obj)
+                if not course:
+                    raise InvalidKeyError('course', 'Wrong course_id')
+                    
+                if email_students:
+                    email_params = get_email_params(course, auto_enroll, secure=request.is_secure())
+                else:
+                    email_params = None
+                validate_email(email)
+                if action == 'enroll':
+                    before, after, _ = enroll_email(
+                        course_id_obj, email, auto_enroll, email_students, email_params, language=language
+                    )
+                else:
+                    before, after = unenroll_email(
+                        course_id_obj, email, email_students, email_params, language=language
+                    )
+
+                result_list.append(
+                    self._make_result_dict_for_identifier(
+                        users_email=users_email, after=after.to_dict(), before=before.to_dict(),
+                    )
+                )
+            except ValidationError:
+                result_list.append(
+                    self._make_result_dict_for_identifier(
+                        users_email=users_email,
+                        error='User with {} users_email has not valid email'.format(users_email),
+                        after={}, before={}
```

Comment:
All things with the code duplication should be created as default parameters of the function.